src/routes/assistants/+page.svelte

```diff
@@ -14,22 +14,36 @@
 	import CarbonArrowUpRight from "~icons/carbon/arrow-up-right";
 	import CarbonEarthAmerica from "~icons/carbon/earth-americas-filled";
 	import CarbonUserMultiple from "~icons/carbon/user-multiple";
+	import CarbonSearch from "~icons/carbon/search";
 	import Pagination from "$lib/components/Pagination.svelte";
 	import { formatUserCount } from "$lib/utils/formatUserCount";
 	import { getHref } from "$lib/utils/getHref";
+	import { debounce } from "$lib/utils/debounce";
 
 	export let data: PageData;
 
 	$: assistantsCreator = $page.url.searchParams.get("user");
 	$: createdByMe = data.user?.username && data.user.username === assistantsCreator;
 
+	const DEBOUNCE_DELAY = 400;
+	let filterInputEl: HTMLInputElement;
+
 	const onModelChange = (e: Event) => {
 		const newUrl = getHref($page.url, {
 			newKeys: { modelId: (e.target as HTMLSelectElement).value },
 			existingKeys: { behaviour: "delete_except", keys: ["user"] },
 		});
 		goto(newUrl);
 	};
+
+	const filterOnName = debounce(async (e: Event) => {
+		const value = (e.target as HTMLInputElement).value;
+		const newUrl = getHref($page.url, { newKeys: { q: value } });
+		await goto(newUrl);
```

Comment:
taking advantage of svelte partial hydration without creating explciit API routes for now